tests/attribute_annotations.pyi

```diff
@@ -47,10 +56,16 @@ field182: Final = os.pathsep
 field183: Final = None
 
 # We *should* emit Y015 for more complex default values
-field19 = [1, 2, 3]  # Y015 Only simple default values are allowed for assignments
-field20 = (1, 2, 3)  # Y015 Only simple default values are allowed for assignments
-field21 = {1, 2, 3}  # Y015 Only simple default values are allowed for assignments
-field22: Final = {"foo": 5}  # Y015 Only simple default values are allowed for assignments
+field221: list[int] = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11]  # Y015 Only simple default values are allowed for assignments
+field222: list[int] = [100000000000000000000000000000]  # Y054 Numeric literals with a string representation >10 characters long are not permitted
+field223: list[int] = [*range(10)]  # Y015 Only simple default values are allowed for assignments
+field224: list[int] = list(range(10))  # Y015 Only simple default values are allowed for assignments
+field225: list[object] = [{}, 1, 2]  # Y015 Only simple default values are allowed for assignments
+field226: tuple[str | tuple[str, ...], ...] = ("foo", ("foo", "bar"))  # Y015 Only simple default values are allowed for assignments
+field227: dict[str, object] = {"foo": {"foo": "bar"}}  # Y015 Only simple default values are allowed for assignments
```

Comment:
I think these are decent examples, why we shouldn't forbid nesting. It's extra code and might occasionally be useful. If it becomes too complex, a manual reviewer can always point this out.

That said, this is not a show stopped.

Reply:
I thought you might say that :)

I guess my worry is that we've added most of our defaults up till now using `stubdefaulter`, which tends to result in a huge PR that we can't really review manually. I wouldn't really want a default such as `{(("foo", "bar"), (1, 2): {("foo", "bar"): {1, 2, 3}}}` added automatically by stubdefaulter in some bizarre third-party library. But I guess that's pretty unlikely; I don't feel too strongly about disallowing nesting.